src/app/blog/category-detail/category-detail.component.ts

```diff
@@ -0,0 +1,45 @@
+import {Component, OnInit} from "@angular/core";
+import {PostService} from "../shared/post.service";
+import {ActivatedRoute, Params} from "@angular/router";
+import {LogFactory} from "../../shared/log.factory";
+import {BlogTitleService} from "../shared/blog-title.service";
+
+@Component({
+  providers: [
+    PostService,
+  ],
+  selector: 'category-detail',
+  templateUrl: './category-detail.component.html',
+  styleUrls: ['./category-detail.component.css']
+})
+export class CategoryDetailComponent implements OnInit {
+
+  constructor(private logFactory: LogFactory,
+              private posts: PostService,
+              private titleService: BlogTitleService,
+              private route: ActivatedRoute) {
+  }
+
+  private logger = this.logFactory.getLog(CategoryDetailComponent.name);
+  private categoryName = '';
+  private postList = [];
+
+  ngOnInit() {
+    let vm = this;
+    vm.queryPostList();
+  }
+
+  queryPostList() {
+    let vm = this;
+    vm.route.params.switchMap(function (params: Params) {
```

Comment:
```suggestion
    vm.route.params.switchMap(params => {
```
1. 总是优先使用箭头函数
2. 函数参数大多数情况下能够推导类型，而单独命名常量变量时建议指定类型